keps/sig-api-machinery/0006-apply.md

```diff
@@ -285,6 +286,70 @@ updated when we know the concrete things changing for beta.
 This will be promoted to GA once it's gone a sufficient amount of time as beta
 with no changes. A KEP update will precede this.
 
+### Upgrade / Downgrade Strategy
+
+With client-side `kubectl apply`, the annotation
+`kubectl.kubernetes.io/last-applied-configuration` tracks ownership for a
+single shared field manager.
+With server-side `kubectl apply --server-side`, the
+`.metadata.managedFields` field tracks ownership for multiple field managers.
+Due to these significant semantic differences, it is not
+possible to accurately sync the intent in the
+`last-applied-configuration` annotation with the `.managedFields` field.
+
+#### Upgrade from Client-Side to Server-Side Apply
+
+When upgrading the API server with server-side apply enabled, then existing objects
+with a `last-applied-configuration` annotation will be upgraded on the first
+use of server-side apply with `kubectl apply --server-side` by
```

Comment:
this implies that when we turn SSA on for everything, we'll actually check for the presence of the annotation and not do the update logic if it is found (or at least not store the result)?

Reply:
Oh no, I'll fix this. We do want to proceed with and store the update logic after reading and converting the annotation.